framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts

```diff
@@ -12,25 +12,281 @@
+		for (let i = 0; i < initValidators.length; i += 1) {
+			const currentValidator = initValidators[i];
+
+			// The blsKeys must be lexigraphically ordered and unique within the array.
+			if (
+				initValidators[i + 1] &&
+				currentValidator.blsKey.compare(initValidators[i + 1].blsKey) > -1
+			) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Validators blsKeys must be unique and lexigraphically ordered'),
+				};
+			}
+
+			if (currentValidator.bftWeight <= BigInt(0)) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Validator bft weight must be greater than 0'),
+				};
+			}
+
+			totalBftWeight += currentValidator.bftWeight;
+		}
+
+		if (totalBftWeight > MAX_UINT64) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(`Validator bft weight must not exceed ${MAX_UINT64}`),
+			};
+		}
+
+		// Minimum certificateThreshold value: floor(1/3 * totalWeight) + 1
+		// Note: BigInt truncates to floor
+		if (certificateThreshold < totalBftWeight / BigInt(3) + BigInt(1)) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate threshold below minimum bft weight '),
+			};
+		}
+
+		// Maximum certificateThreshold value: total bft weight
+		if (certificateThreshold > totalBftWeight) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate threshold above maximum bft weight'),
+			};
+		}
+
+		return {
+			status: VerifyStatus.OK,
+		};
+	}
+
+	public async execute(context: CommandExecuteContext<SidechainRegistrationParams>): Promise<void> {
+		const {
+			transaction,
+			params: { certificateThreshold, initValidators, genesisBlockID, name },
+			getStore,
+		} = context;
+
+		const networkID = hash(Buffer.concat([genesisBlockID, transaction.senderAddress]));
+
+		// Add an entry in the chain substore
+		const chainSubstore = getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA);
+
+		// Find the latest chainID from db
+		const start = intToBuffer(0, 4, BIG_ENDIAN);
+		const end = intToBuffer(MAX_UINT32, 4, BIG_ENDIAN);
+		const chainIDs = await chainSubstore.iterate({ start, end, limit: 1, reverse: true });
+		if (!chainIDs.length) {
+			throw new Error('No existing entries found in chain store');
+		}
+		const chainID = chainIDs[0].key.readUInt32BE(0) + 1;
+		const chainIDBuffer = intToBuffer(chainID, 4, BIG_ENDIAN);
+
+		await chainSubstore.setWithSchema(
+			chainIDBuffer,
+			{
+				name,
+				networkID,
+				lastCertificate: {
+					height: 0,
+					timestamp: 0,
+					stateRoot: EMPTY_HASH,
+					validatorsHash: computeValidatorsHash(initValidators, certificateThreshold),
+				},
+				status: 0,
+			},
+			chainAccountSchema,
+		);
+
+		// Add an entry in the channel substore
+		const channelSubstore = getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHANNEL_DATA);
+		await channelSubstore.setWithSchema(
+			chainIDBuffer,
+			{
+				inbox: { root: EMPTY_HASH, appendPath: [], size: 0 },
```

Comment:
here also default value of bytes is empty bytes? same for other EMPTY_HASH property

Reply:
Default value for `inbox.root` is `EMPTY_HASH` and not `EMPTY_BYTES` as per LIP 0045

Reply:
ok! i missed that part 👍  same for above